File: backend/app/schemas/class_subject_requirement.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class ClassSubjectRequirementBase(BaseModel):
    semester_id: int
    class_id: int
    subject_id: int
    weekly_periods: int
    required_teacher_id: int | None = None
    required_room_type: str | None = None
    consecutive_limit: int | None = None


class ClassSubjectRequirementCreate(ClassSubjectRequirementBase):
    pass


class ClassSubjectRequirementRead(ClassSubjectRequirementBase):
    id: int
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class ClassSubjectRequirementUpdate(BaseModel):
    semester_id: int | None = None
    class_id: int | None = None
    subject_id: int | None = None
    weekly_periods: int | None = None
    required_teacher_id: int | None = None
    required_room_type: str | None = None
    consecutive_limit: int | None = None

File: frontend/src/types/classSubjectRequirement.ts
// Mirrors backend/app/schemas/class_subject_requirement.py in full.

export interface ClassSubjectRequirement {
  id: number;
  semester_id: number;
  class_id: number;
  subject_id: number;
  weekly_periods: number;
  required_teacher_id: number | null;
  required_room_type: string | null;
  consecutive_limit: number | null;
  created_at: string;
  updated_at: string;
}

export interface ClassSubjectRequirementCreate {
  semester_id: number;
  class_id: number;
  subject_id: number;
  weekly_periods: number;
  required_teacher_id?: number | null;
  required_room_type?: string | null;
  consecutive_limit?: number | null;
}

export interface ClassSubjectRequirementUpdate {
  semester_id?: number;
  class_id?: number;
  subject_id?: number;
  weekly_periods?: number;
  required_teacher_id?: number | null;
  required_room_type?: string | null;
  consecutive_limit?: number | null;
}
